### PR TITLE
DBZ-1834 Support dots in table names

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-1834

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Hey @jpechane, test failure looks related?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java [hunk 1]
```diff
@@ -215,7 +215,8 @@ public String parseName(MySqlParser.UidContext uidContext) {
     public TableId parseQualifiedTableId(MySqlParser.FullIdContext fullIdContext) {
         String fullTableName = fullIdContext.getText();
         int dotIndex;
-        if ((dotIndex = fullTableName.indexOf(".")) > 0) {
+        // The quoted table name that contains dot means that it is just a part of the name of the table
+        if (!isQuoted(fullTableName) && ((dotIndex = fullTableName.indexOf(".")) > 0)) {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java]: Couldn't we encounter something like ``inventory.`My.Table` ``?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java]: @gunnarmorling Just fixed

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java [hunk 2]
```diff
@@ -213,15 +213,39 @@ public String parseName(MySqlParser.UidContext uidContext) {
...
+            if (isQuote(c)) {
+                // Opening quote
+                if (lastQuote == EMPTY) {
+                    lastQuote = c;
+                }
+                // Closing quote
+                else if (lastQuote == c) {
+                    lastQuote = EMPTY;
+                }
+                // Quote that is part of name
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java]: Do we have any test for this case actually? How would escaping of the quote look like?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java]: @gunnarmorling Added more comprehensive test and escaping support

**Comment by gunnarmorling** [comment]
Applied; thx!

**Pull request merged**